pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -460,6 +461,52 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		volumes[volume.Name] = volume.DeepCopy()
 	}
 
+	dedicatedThreads := 0
+	autoThreads := 0
+	useIOThreads := false
+	threadPoolLimit := 1
+
+	if vmi.Spec.Domain.IOThreadsPolicy != nil {
+		useIOThreads = true
+
+		if (*vmi.Spec.Domain.IOThreadsPolicy) == "auto" {
+			threadPoolLimit = 2
+			if vmi.Spec.Domain.CPU != nil {
+				threadPoolLimit = 2 * int(vmi.Spec.Domain.CPU.Cores)
```

Comment:
I think we should be looking at `spec.domain.resources.requests.cpu` and `spec.domain.resources.limits.cpu` to get the number of vcpu. `spec.domain.cpu.cores` just sets the guest topology which doesn't have to match actual vcpus